symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/RealTimeEventListener.java

```diff
@@ -1,126 +1,173 @@
-     * Called when a ROOMREACTIVATED event is received.
-     * @param initiator Event initiator.
-     * @param event Room Reactivated payload.
-     */
-    default void onRoomReactivated(V4Initiator initiator, V4RoomReactivated event) {}
-
-    /**
-     * Called when an USERREQUESTEDTOJOINROOM event is received.
-     * @param initiator Event initiator.
-     * @param event User Requested To Join Room payload.
-     */
-    default void onUserRequestedToJoinRoom(V4Initiator initiator, V4UserRequestedToJoinRoom event) {}
-
-    /**
-     * Called when an USERJOINEDROOM event is received.
-     * @param initiator Event initiator.
-     * @param event User Joined Room payload.
-     */
-    default void onUserJoinedRoom(V4Initiator initiator, V4UserJoinedRoom event) {}
-
-    /**
-     * Called when an USERLEFTROOM event is received.
-     * @param initiator Event initiator.
-     * @param event User Left Room payload.
-     */
-    default void onUserLeftRoom(V4Initiator initiator, V4UserLeftRoom event) {}
-
-    /**
-     * Called when a ROOMMEMBERPROMOTEDTOOWNER event is received.
-     * @param initiator Event initiator.
-     * @param event Room Member Promoted To Owner payload.
-     */
-    default void onRoomMemberPromotedToOwner(V4Initiator initiator, V4RoomMemberPromotedToOwner event) {}
-
-    /**
-     * Called when a ROOMMEMBERDEMOTEDFROMOWNER event is received.
-     * @param initiator Event initiator.
-     * @param event Room Member Demoted From Owner payload.
-     */
-    default void onRoomMemberDemotedFromOwner(V4Initiator initiator, V4RoomMemberDemotedFromOwner event) {}
-
-    /**
-     * Called when a CONNECTIONREQUESTED event is received.
-     * @param initiator Event initiator.
-     * @param event Connection Requested payload.
-     */
-    default void onConnectionRequested(V4Initiator initiator, V4ConnectionRequested event) {}
-
-    /**
-     * Called when a CONNECTIONACCEPTED event is received.
-     * @param initiator Event initiator.
-     * @param event Connection Accepted payload.
-     */
-    default void onConnectionAccepted(V4Initiator initiator, V4ConnectionAccepted event) {}
-
-    /**
-     * Called when a MESSAGESUPPRESSED event is received.
-     * @param initiator Event initiator.
-     * @param event Message Suppressed payload.
-     */
-    default void onMessageSuppressed(V4Initiator initiator, V4MessageSuppressed event) {}
-
-    /**
-     * Called when a SYMPHONYELEMENTSACTION event is received.
-     * @param initiator Event initiator.
-     * @param event Symphony Elements Action payload.
-     */
-    default void onSymphonyElementsAction(V4Initiator initiator, V4SymphonyElementsAction event) {}
+  /**
+   * Check if the event is accepted to be handled.
+   *
+   * @param event   Event to be verified.
+   * @param config  Bdk config to be used in filtering the event out.
+   * @return the event is accepted or not
+   */
+  default boolean isAcceptingEvent(V4Event event, BdkConfig config) {
```

Comment:
Do we really need to pass the whole `BdkConfig` object here? Maybe just the bot `username` is enough

Reply:
This might prevent future usages from accessing other information than the username to decide if the event needs to be filtered or not but I agree that BdkConfig feels off to pass here (and I don't have better ideas except introducing a specific object for it)

Reply:
so I guess for the moment, we should annotate the method with apiguardian`EXPERIMENTAL`?